nemo/lightning/pytorch/strategies/megatron_strategy.py

```diff
@@ -849,12 +844,14 @@ def load_checkpoint(self, checkpoint_path: Union[str, Path], selective_restore:
 
         if HAVE_MODELOPT:
             # If present, first restore and modify the model according to the ModelOpt state.
+            # Avoid quantizers being added to teacher model if model is a distillation model.
             core_model = unwrap_model(self.megatron_parallel)
-            restore_sharded_modelopt_state(
-                [core_model],
-                ckpt_to_weights_subdir(checkpoint_path, is_saving=False),
-            )
-            if ModeloptStateManager.is_converted(core_model):
+            with core_model.hide_teacher_model() if hasattr(core_model, "hide_teacher_model") else nullcontext():
```

Comment:
This change enables us to load quantized student for distillation (QAT) otherwise teacher will also have quantizers added but no state dict for it.

This happens because quantization config has regex `*output_layer*` causing `_teacher_model.*output_layer*` to also match